test/libsolidity/SemanticTest.cpp

```diff
@@ -78,112 +78,141 @@ SemanticTest::SemanticTest(string const& _filename, langutil::EVMVersion _evmVer
+					test.setContractABI(m_compiler.contractABI(m_compiler.lastContractName()));
+				}
 			}
-			_stream << endl;
-			AnsiColorized(_stream, _formatted, {BOLD, CYAN}) << _linePrefix << "Obtained result:" << endl;
-			for (auto const& test: m_tests)
+
+			if (success && !m_runWithYul && compileViaYul)
 			{
-				ErrorReporter errorReporter;
-				_stream << test.format(errorReporter, _linePrefix, true, _formatted) << endl;
-				_stream << errorReporter.format(_linePrefix, _formatted);
+				m_compileViaYulCanBeSet = true;
+				AnsiColorized(_stream, _formatted, {BOLD, YELLOW}) << _linePrefix << endl << _linePrefix
+					<< "Test can pass via Yul and marked with compileViaYul: false." << endl;
+				return TestResult::Failure;
 			}
-			AnsiColorized(_stream, _formatted, {BOLD, RED}) << _linePrefix << endl << _linePrefix
-				<< "Attention: Updates on the test will apply the detected format displayed." << endl;
-			if (compileViaYul && m_runWithoutYul)
+
+			if (!success && (m_runWithYul || !compileViaYul))
 			{
-				_stream << _linePrefix << endl << _linePrefix;
-				AnsiColorized(_stream, _formatted, {RED_BACKGROUND}) << "Note that the test passed without Yul.";
+				AnsiColorized(_stream, _formatted, {BOLD, CYAN}) << _linePrefix << "Expected result:" << endl;
+				for (auto const& test: m_tests)
+				{
+					ErrorReporter errorReporter;
+					_stream << test.format(errorReporter, _linePrefix, false, _formatted) << endl;
+					_stream << errorReporter.format(_linePrefix, _formatted);
+				}
 				_stream << endl;
+				AnsiColorized(_stream, _formatted, {BOLD, CYAN}) << _linePrefix << "Obtained result:" << endl;
+				for (auto const& test: m_tests)
+				{
+					ErrorReporter errorReporter;
+					_stream << test.format(errorReporter, _linePrefix, true, _formatted) << endl;
+					_stream << errorReporter.format(_linePrefix, _formatted);
+				}
+				AnsiColorized(_stream, _formatted, {BOLD, RED}) << _linePrefix << endl << _linePrefix
+					<< "Attention: Updates on the test will apply the detected format displayed." << endl;
+				if (compileViaYul && m_runWithoutYul)
+				{
+					_stream << _linePrefix << endl << _linePrefix;
+					AnsiColorized(_stream, _formatted, {RED_BACKGROUND})
+						<< "Note that the test passed without Yul.";
+					_stream << endl;
+				}
+				else if (!compileViaYul && m_runWithYul)
+					AnsiColorized(_stream, _formatted, {BOLD, YELLOW}) << _linePrefix << endl << _linePrefix
+						<< "Note that the test also has to pass via Yul." << endl;
+				return TestResult::Failure;
 			}
-			else if (!compileViaYul && m_runWithYul)
-				AnsiColorized(_stream, _formatted, {BOLD, YELLOW}) << _linePrefix << endl << _linePrefix
-					<< "Note that the test also has to pass via Yul." << endl;
-			return TestResult::Failure;
+		}
+		catch (boost::exception const& _e)
+		{
+			if (compileViaYul && !m_runWithYul)
+				continue;
+
+			throw;
+		}
+		catch (std::exception const& _e)
+		{
+			if (compileViaYul && !m_runWithYul)
+				continue;
```

Comment:
the `_e` parameter is unused and causes an error

Reply:
I removed it now, will rebase, check if additional tests are activated and push